packages/website/components/account/filesManager/cellRendererComponents/uploadStatusCell.js

```diff
@@ -0,0 +1,41 @@
+import { BsFillInfoCircleFill } from 'react-icons/bs';
+
+import Tooltip from 'ZeroComponents/tooltip/tooltip';
+
+export const PinStatus = {
+  PINNED: 'Pinned',
+  PINNING: 'Pinning',
+  PIN_QUEUED: 'PinQueued',
+  QUEUING: 'Queuing...',
+};
+
+/**
+ * @type {import('react').FC}
+ * @param {object} props
+ * @returns
+ */
+function uploadStatusTableRenderer({ pins, statusMessages }) {
+  if (!pins) {
+    return <span></span>;
+  }
+
+  const status = Object.values(PinStatus).find(status => pins.some(pin => status === pin.status)) || PinStatus.QUEUING;
+
+  const statusTooltips = {
+    [PinStatus.QUEUING]: statusMessages.queuing,
+    [PinStatus.PIN_QUEUED]: statusMessages.pin_queued,
+    [PinStatus.PINNING]: statusMessages.pinning,
+    [PinStatus.PINNED]: statusMessages.pinned.replace('*numberOfPins*', `${pins.length}`),
```

Comment:
Likely outside of the context of this ticket, but I wonder why we aren't using string templates here? 🤔

Reply:
Fair point. Happy to follow the pattern.